csw-location/src/main/scala/csw/services/location/internal/LocationServiceImpl.scala

```diff
@@ -103,107 +76,67 @@ private[location] class LocationServiceImpl(cswCluster: CswCluster) extends Loca
     case NotFound(AllServices.Key, _)     ⇒ List.empty
     case _                                => throw RegistrationListingFailed
   }
 
-  /**
-    * List all locations from `LWWMap` and complete the `Future` with `Location` values filtered on `ComponentType`.
-    *
-    * A `Future` is returned with empty list if no constant key is found for `LWWMap` or no locations are registered
-    * against the given `ComponentType`.
-    *
-    * The returned `Future` will fail if list from `LWWMap` fails
-    */
+  //List all locations registered for the given componentType
   def list(componentType: ComponentType): Future[List[Location]] = async {
     await(list).filter(_.connection.componentId.componentType == componentType)
   }
 
-  /**
-    * List all locations from `LWWMap` and complete the `Future` with `Location` values filtered on `Hostname`.
-    *
-    * A `Future` is returned with empty list if no constant key is found for `LWWMap` or no locations are registered
-    * against the given `Hostname`.
-    *
-    * The returned `Future` will fail if list from `LWWMap` fails
-    */
+  //List all locations registered with the given hostname
   def list(hostname: String): Future[List[Location]] = async {
     await(list).filter(_.uri.getHost == hostname)
   }
 
-  /**
-    * List all locations from `LWWMap` and complete the `Future` with `Location` values filtered on `ConnectionType`.
-    *
-    * A `Future` is returned with empty list if no constant key is found for `LWWMap` or no locations are registered
-    * against the given `ComponentType`.
-    *
-    * The returned `Future` will fail if list from `LWWMap` fails
-    */
+  //List all locations registered with the given connection type
   def list(connectionType: ConnectionType): Future[List[Location]] = async {
     await(list).filter(_.connection.connectionType == connectionType)
   }
 
-  /**
-    * Creates an `ActorRef` that subscribes for `Changed` messages for a given `Connection` from `LWWRegister` and pass
-    * it to [[akka.stream.scaladsl.Source]].
-    *
-    * The `Source` will then map the `Changed` event to [[csw.services.location.models.LocationUpdated]]
-    * if the `LWWRegister` contains the `Location` against `Connection`
-    *
-    * or to [[csw.services.location.models.LocationRemoved]] if there is no value against `Connection`.
-    *
-    * Un-track a given connection using [[akka.stream.KillSwitch]]
-    *
-    */
+  //Track the status of given connection
   def track(connection: Connection): Source[TrackingEvent, KillSwitch] = {
+    //Create a CRDT key from this connection
     val service = new Registry.Service(connection)
+    //Get a source from an actor so that messages send to the actor is put in the stream
     val source = Source.actorRef[Any](256, OverflowStrategy.dropHead).mapMaterializedValue {
+      //When the stream starts flowing, the actor is subscribed to the replicator to get all events for the connection key
```

Comment:
This still does not explain it well, look at the docs from Json's branch for better wording.. like all messages sent to the actor will appear on the stream. .which is the main point.